parsl/tests/test_python_apps/test_garbage_collect.py

```diff
@@ -17,14 +17,13 @@ def test_garbage_collect():
     """
     x = slow_double(slow_double(10))
 
-    if x.done() is False:
-        assert parsl.dfk().tasks[x.tid]['app_fu'] == x, "Tasks table should have app_fu ref before done"
+    assert parsl.dfk().tasks[x.tid]['app_fu'] == x, "Tasks table should have app_fu ref before done"
```

Comment:
theres a slight race condition here that the x.done() is trying to gloss over: x could have executed and been garbage collected by the time this line is reached, if there's sufficient delay here... (that race condition also existed in different form with the previous `if x.done() is False` test...)

there's no guarantee apart from hoping that time.sleep lasts long enough that this assertion really does happen before the x future is done.

what might be better for this test is running in-process using the thread local executor, and using an Event to control the completion of the test app

Reply:
:face_palm: This was precisely the snippet I shared with you in Slack, but then realized would not work for anything other than threads (and have thus deleted the shared snippet!).

It sounds like you're suggesting to only strictly run the test in a local configuration then?